CMakeLists.txt

```diff
@@ -118,3 +118,16 @@ install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
 # Enable unit testing by including the `test` subdirectory.
 enable_testing()
 add_subdirectory(test)
+
+###############
+## UNINSTALL ##
+###############
+
+# Add uninstall target
+configure_file(
+  "${CMAKE_CURRENT_SOURCE_DIR}/cmake/cmake_uninstall.cmake.in"
+  "${CMAKE_CURRENT_BINARY_DIR}/cmake/cmake_uninstall.cmake"
+  IMMEDIATE @ONLY)
+add_custom_target(uninstall
+  "${CMAKE_COMMAND}" -P
+  "${CMAKE_CURRENT_BINARY_DIR}/cmake/cmake_uninstall.cmake")
```

Comment:
One corner case we might need to consider would be uninstalling the package installed by a package manager. We probably don't want to remove the files using `uninstall.sh`. For that case, we should have a flag whether it installs `uninstall.sh` and set false when the package is installed by the package manager.